subsys/net/lib/nrf_cloud/coap/src/nrfc_dtls.c

```diff
@@ -167,19 +168,30 @@ int nrfc_dtls_setup(int sock)
 
 	int session_cache = TLS_SESSION_CACHE_ENABLED;
 
+	LOG_DBG("  TLS session cache: %d", session_cache);
 	err = setsockopt(sock, SOL_TLS, TLS_SESSION_CACHE, &session_cache, sizeof(session_cache));
 	if (err) {
 		LOG_ERR("Failed to enable session cache, errno: %d", -errno);
 		err = -errno;
 	}
 
+	keepopen_supported = false;
 	if (IS_ENABLED(CONFIG_NRF_CLOUD_COAP_KEEPOPEN)) {
 		err = setsockopt(sock, SOL_SOCKET, SO_KEEPOPEN, &(int){1}, sizeof(int));
 		if (err) {
-			LOG_ERR("Failed to set SO_KEEPOPEN: %d", -errno);
-			err = -errno;
+			if (errno != EINVAL) {
+				LOG_ERR("Failed to set SO_KEEPOPEN: %d", -errno);
```

Comment:
when reading `keepopen_supported`, i thought "supported by the modem", as in, it should work.
but this handling seems to be more like `keepopen_enabled`.

for this error case, does it mean that keepopen _is_ supported, but failed for whatever reason?

Reply:
I am not sure what all possible errno values are for this socket option and what they all mean. I can probably just simplify this to assume any error indicates we can't use SO_KEEPOPEN though.

Reply:
> I can probably just simplify this to assume any error indicates we can't use SO_KEEPOPEN though.

Agreed.